src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -213,14 +236,14 @@ public async Task<SyntaxNode> GetSyntaxRootAsync(CancellationToken cancellationT
         /// on the UI thread).  Right now, the only feature this is for is Outlining as VS will
         /// block on that feature from the UI thread when a document is opened.
         /// </summary>
-        internal SyntaxNode GetSyntaxRootSynchronously(CancellationToken cancellationToken)
+        internal SyntaxNode? GetSyntaxRootSynchronously(CancellationToken cancellationToken)
         {
             if (!this.SupportsSyntaxTree)
             {
                 return null;
             }
 
-            var tree = this.GetSyntaxTreeSynchronously(cancellationToken);
+            var tree = this.GetSyntaxTreeSynchronously(cancellationToken)!;
```

Comment:
💡 This can also be rewritten to use a call followed by a null check.

Reply:
My rule was to use suppression for "the cases where it's immediately obvious from context"; in this case, it's obvious we've already handled the non-supporting-syntax-tree case.